tests/weighted_layers_test.py

```diff
@@ -0,0 +1,192 @@
+    return [layer_inputs, self.xi(None)]
+
+  def test_output_shape(self):
+    """checks if the shape of the output tensor is correct."""
+    w_d_conv = self._create_default_depth_conv()
+    input_shape = (32, 64, 64, self.input_channel)
+    inputs = self._get_default_inputs(input_shape)
+    output = w_d_conv(inputs)
+    expected_shape = (32, 64, 64, self.input_channel*self.depth_multiplier)
+    self.assertAllEqual(expected_shape, output.shape)
+
+  def test_output_value(self):
+    """checks if the value of the output tensor is correct."""
+    w_d_conv = self._create_default_depth_conv()
+    w_d_conv.activation = None
+    input_shape = (32, 16, 16, self.input_channel)
+    inputs = self._get_default_inputs(input_shape)
+    w_d_output = w_d_conv(inputs)
+
+    # the output of weighted convolution should be same as linear combination of
+    # outputs of regular convolution with template weights in case when no
+    # activation is used.
+    expected_output = tf.zeros_like(w_d_output)
+    conv = tf.keras.layers.DepthwiseConv2D(
+        depth_multiplier=self.depth_multiplier, activation=None,
+        padding=self.padding, kernel_size=self.kernel_size)
+    conv.build(input_shape)
+    for t in range(self.num_templates):
+      conv.depthwise_kernel = self.kernel[t]
+      conv.bias = self.bias[t]
+      conv_out = conv(inputs[0])
+      expected_output += inputs[1][t]*conv_out
+    self.assertAllClose(expected_output, w_d_output, rtol=1e-05)
+
+class WeightedBatchNormalizationTest(tf.test.TestCase):
+  """"WeightedBatchNormalizationSeparate test class."""
+  def setUp(self):
+    """Sets default parameters."""
+    self.num_templates = 10
+    self.input_channels = 40
+    self.gamma_template = np.random.rand(self.num_templates,
+                                         self.input_channels).astype(np.float32)
+    self.beta_template = np.random.rand(self.num_templates,
+                                        self.input_channels).astype(np.float32)
+    self.beta_init = tf.constant_initializer(self.beta_template)
+    self.gamma_init = tf.constant_initializer(self.gamma_template)
+    self.xi_initializer = tf.random_uniform_initializer(minval=0.0, maxval=1.0)
+    self.xi = MixtureWeight(num_templates=self.num_templates,
+                            initializer=self.xi_initializer)
+
+  def test_output_shape(self):
+    """checks if the output shape is same as input shape."""
+    input_shape = (256, 16, 16, self.input_channels)
+    inputs = tf.random.normal(input_shape)
+    bn = WeightedBatchNormalizationSeparate(num_templates=self.num_templates,
+                                            gamma_initializer=self.gamma_init,
+                                            beta_initializer=self.beta_init)
+    outputs = bn([inputs, self.xi(None)], training=True)
+    self.assertAllEqual(input_shape, outputs.shape)
+
+  def test_output_moments(self):
+    """checks if the moments of the output tensor match to the value of mixture
```

Comment:
Great test! Side note: I would try to shrink the function docstring so that it fits in a single line.